### ISSUE TITLE
Does not work on Gnome 3.33.91

### ISSUE THREAD

**Issue by silviobierman** [passingthru67/workspaces-to-dock#177]
The Gnome version on the Ubuntu 19.10 development version was bumped to 3.33.91 and now Workspaces-to-Dock no longer works. Working with plain vanilla Gnome now :-((

**Comment by reuixiy** [comment]
GNOME Shell 3.34.0 on Arch Linux, too...

**Comment by silviobierman** [comment]
Yes. And the same is true for Fedora 31 (also Gnome 3.34). But unfortunately the author has announced that he lacks time to work on the plugin so I guess we just have to be patient. Hopefully he will come round to this eventually or someone else steps in and helps him.

**Comment by Orcuslc** [comment]
Same for GNOME 3.34 on Arch Linux. Could cause GNOME Shell to freeze.

**Comment by passingthru67** [comment]
I finally started work on Gnome 3.34 compatibility today. While vanilla Gnome is certainly usable, I find WtD and a few other extensions make my work flow a little easier.

BTW, I'm still hoping to turn over the ownership of this project to a new developer. I'd still be around for the occasional assist or pull request. But, my spare time just isn't what it used to be.

**Comment by silviobierman** [comment]
That is great to hear. I find the term "usable" quite generous. I can not understand why Dash-to-Dock and Workspaces-to-Dock are not part of Gnome-shell. They make an awkward desktop into one of the best I have worked with. I will be available for any testing you require.

**Comment by passingthru67** [comment]
I just uploaded the gnome-3.34-dev (https://github.com/passingthru67/workspaces-to-dock/tree/gnome-3.34-dev) branch for testing. It's still a work in progress so expect bugs.

Known issues:
- Left & Top position slideout is glitchy
- Gnome overview mode shows shrunken app thumbnail windows. Clears up after initially entering overview mode

**Comment by silviobierman** [comment]
I have installed this and did some testing. It kind of works (slides out initially) but after sliding out/in a couple of times (without actually switching workspaces) my Gnome shell crashes and logs me out. I did not succeed in getting an error message via ALT-F2 lg because the crash happens too quickly. So I am afraid I have no usefull information for you. I use right position, extended to full height.

Is there another place where I can find error messages? Some log file or something?

**Comment by passingthru67** [comment]
You can use the journalctl --since and --until options like below ..
 
journalctl --since "2019-09-26 08:15:00" --until "2019-09-26 08:17:00" > journal.txt
(don’t forget to replace the date/time with the date & time of the issue)

I’m hoping the logs will show what’s going on.

**Comment by silviobierman** [comment]
Here is my log of one minute where I installed W2D, rebooted, logged in, caused the crash, rebooted again. As the log shows the crash resulted in disabling shell extensions. I uninstalled W2D and enabled the extensions and ran the journalctl. 
[Uploading journalctl.txt…]()

**Comment by silviobierman** [comment]
Sorry. Closed this by accident.

**Comment by silviobierman** [comment]
BTW: Fedora 31 so Wayland...

**Comment by passingthru67** [comment]
@silviobierman Did you post the journal log in the chat? I don't see it attached anywhere.

**Comment by silviobierman** [comment]
[journalctl.txt](https://github.com/passingthru67/workspaces-to-dock/files/3659780/journalctl.txt)

This seems to have worked...

**Comment by passingthru67** [comment]
@silviobierman Would you do a quick test with Gnome on Xorg to verify the problem persists?

**Comment by silviobierman** [comment]
I just tried. Apparently here is no crash but sliding in/out repeatedly still results in erratic behavior. The whole screen goes black for a second and then the shell reappears. But I am still logged in and W2D still works. So there IS a difference between Wayland and Xorg but it does not work flawlessly on Xorg. ALT-F2 lg say no errors emitted...

**Comment by passingthru67** [comment]
@silviobierman Are you testing with other extensions enabled? I haven't checked compatibility with Dash-To-Dock yet so there may be issues because of all the code changes.

**Comment by reuixiy** [comment]
Hi @silviobierman,

I have tested that branch on `GNOME 3.34 on Xorg, Arch Linux` and observed the issues you said. Also, I found that the title of the window is not in the correct position.

![2019-09-27-10-32-53](https://user-images.githubusercontent.com/28993043/65738154-7d0ad180-e113-11e9-98a6-27e81a0212c4.png)

When I turn the extension on, I only get the following logs:

```
Sep 27 10:35:01 arch gnome-shell[8577]: st_widget_get_theme_node called on the widget [0x559571df06c0 StBin:insensitive] which is not in the stage.
Sep 27 10:35:01 arch gnome-shell[8577]: st_widget_get_theme_node called on the widget [0x559571df06c0 StBin:insensitive] which is not in the stage.
```

I hope it will be useful to you and thank you for your great work! :+1:

**Comment by silviobierman** [comment]
@passingthru67: Yes, I have D2D installed. Will test without it this evening.

**Comment by passingthru67** [comment]
Today's update fixes:
1) Left/Top position glitchiness
2) App window title labels
3) Dash-To-Dock compatibility

**Comment by silviobierman** [comment]
I have tested the current state on both Wayland and Xorg with D2D enabled and disabled. The crash is still there on Wayland and it still gives a temporary black screen on Xorg. D2D does not seem to have an influence on this.
I simply show/hide by touching the right side of the screen repeatedly. Six or eight times does it for me. I have disabled the "pressure" option and simple show when the mouse hits the right side.

**Comment by passingthru67** [comment]
Would you post a dump of your WtD configuration? Just type the following in the command line ..

dconf dump /org/gnome/shell/extensions/workspaces-to-dock/ > dconfdump.txt

**Comment by dpriedel** [comment]
Hi
I have been running the version from 3.34-dev for a few hours now with no problems.

Using Manjaro with latest Gnome updates from testing repositories.

Thanks for this !!

**Comment by reuixiy** [comment]
Yep, it works well! :heart: Now I only got one issue:

> Gnome overview mode shows shrunken app thumbnail windows. Clears up after initially entering overview mode.

**Comment by silviobierman** [comment]
@passingthru67: Sure, here is the content:

[/]
background-opacity=1.0
customize-height=true
customize-height-option=1
customize-thumbnail=true
dock-fixed=false
intellihide=false
require-pressure-to-show=false
thumbnail-size=0.16600000000000001
workspace-captions=false

**Comment by silviobierman** [comment]
@passingthru67: I just updated my Fedora31 (still pre-release) and got a new mutter among other things. I am now unable to reproduce the issue so I suspect it was related to some bug in my environment. I hope I did not cause you to run around in circles.

Thanks a lot again for the good work. I love W2D and would not use Gnome shell without it (and D2D).

**Comment by passingthru67** [comment]
@silviobierman No problem.

**Comment by passingthru67** [comment]
Today's update fixes the overview mode shrunken geometry issue. I've gone ahead and merged the gnome-3.34-dev into the master branch.

If no more bugs are reported, I will upload this next version to the extension.gnome.org site this week.

**Comment by silviobierman** [comment]
This is working fine for me. Thanks again.

**Comment by ysolis** [comment]
Archlinux with Gnome 3.34 ... fails with Xorg and Wayland

journalctl log related

```
banshee gnome-shell[3792]: JS WARNING: [/home/yonsy/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js 2269]: reference to undefined property "_thumbnailsBox"
Oct 02 16:44:27 banshee gnome-shell[3792]: JS ERROR: TypeError: this._thumbnailsBox is undefined
                                           _updateSize@/home/yonsy/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js:2269:13
                                           _resetPosition@/home/yonsy/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js:2425:9
                                           _refreshThumbnails@/home/yonsy/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js:2452:9
                                           _overrideGnomeShellFunctions/Layout.LayoutManager.prototype._updateRegions@/home/yonsy/.local/share/gnome-shell/extensions/workspaces-to-dock@passingthru67.gmail.com/dockedWorkspaces.js:720:21

```

**Comment by passingthru67** [comment]
@ysolis Did you install the extension from the master branch?

**Comment by ysolis** [comment]
> @ysolis Did you install the extension from the master branch?

I installed previously from the extension gnome website... yesterday update from the master branch... zero problems until now.
